WordPress/openverse-frontend
# Fix header items not fitting in

## Fixes


Fixes #711 by @zackkirda

## Description


Currently, the mobile header uses `flex-wrap` to wrap the elements and push the search bar to the second row. This prevents us from using the normal `flex` mechanism to ensure that the other row fits into the screen.
This PR sets the width of the search bar using the `CSS` calc property to calculate the width that's left after buttons and paddings. It also removes the margins from the header menu buttons and handles the spacing from the parent (header) using `gap` property.
<img width="429" alt="Screen Shot 2022-01-26 at 9 08 05 AM" src="https://user-images.githubusercontent.com/15233243/151116502-caa9b765-c606-4ccc-93c6-ad5ad9a7b48c.png">

This works fine until the width of 379px. Below that, because the width of the content switcher is hard-coded to 5rem (90px), there is not enough width in the header to fit all components, so the filter button goes to the second line:

<img width="394" alt="Screen Shot 2022-01-26 at 9 08 20 AM" src="https://user-images.githubusercontent.com/15233243/151116724-8f373fe1-06b3-42ee-a023-fdc3f2b92ff5.png">

Another problem I have is with the inner pages on smaller widths, as there is no Figma mockup showing the header on the inner pages. This is what it currently looks like:
<img width="423" alt="Screen Shot 2022-01-26 at 9 40 45 AM" src="https://user-images.githubusercontent.com/15233243/151116898-37da18ee-e183-4bf3-9502-1e9ebb827851.png">


@panchovm, I have 2 questions:
1. What should I do for the widths lower than 394px?
2. Does the inner page header look OK?


## Testing Instructions

Run the app, and try decreasing the screen size down to 350px (or lower?). Try different pages, and with the header scrolled/ not scrolled. All element should fit into one row, or two rows on a search page when the header hadn't been scrolled.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
